app/controllers/api_controller/reports.rb

```diff
@@ -13,5 +13,29 @@ def show_reports
       end
       show_generic(:reports)
     end
+
+    def run_resource_reports(_type, id, _data)
+      report = MiqReport.find(id)
+      report.queue_generate_table do |task, report_result|
```

Comment:
can you update this so we don't need to update the queue_generate_table method ?